PowerUp-2018/src/com/spikes2212/robot/SubsystemComponents.java

```diff
@@ -39,6 +41,21 @@
 		public static final Encoder LEFT_ENCODER = new Encoder(RobotMap.DIO.DRIVE_LEFT_ENCODER_A,
 				RobotMap.DIO.DRIVE_LEFT_ENCODER_B);
 		public static final Gyro GYRO = new ADXRS450_Gyro();
+		public static final Consumer<Double> LEFT_CONTROL = limitSpeed(LEFT_MOTOR::set);
+		public static final Consumer<Double> RIGHT_CONTROL = limitSpeed(RIGHT_MOTOR::set);
+
+		private static Consumer<Double> limitSpeed(Consumer<Double> speedConsumer) {
+			/*
+			 * n = 1 (max voltage), m = (max limit - 1) / 4 -> top speed =
+			 * position * (max limit - 1) / 4 + 1.

```

Comment:
create a new command that overrides the driveArcade command, and set it as the default command for the drivetrain.

Reply:
why? this is why we get a Consumer and not speedControler